asistente_ladm_col/lib/geometry.py

```diff
@@ -721,7 +721,14 @@ def build_boundaries(self, boundary_layer):
         :return: QgsVectorLayer
         """
         params = {'boundaries': boundary_layer, 'native:refactorfields_2:built_boundaries': 'TEMPORARY_OUTPUT'}
-        return processing.run("model:Build_Boundaries", params)['native:refactorfields_2:built_boundaries']
+        build_boundary_layer = processing.run("model:Build_Boundaries", params)['native:refactorfields_2:built_boundaries']
+
+        # For CTM12 the output layer remains without projection. The projection of the input layer is assigned
```

Comment:
Is it necessary? 

Wasn't it only an issue for unit tests?

If you really need to set the CRS here, just use `layer.setCrs()`.